app/src/main/java/com/sopt/peekabookaos/presentation/barcodeScanner/BarcodeScannerActivity.kt

```diff
@@ -0,0 +1,203 @@
+                    val camera = cameraProvider.bindToLifecycle(
+                        this,
+                        cameraSelector,
+                        preview,
+                        textBarcodeAnalyzer
+                    )
+                    cameraControl = camera.cameraControl
+                    cameraInfo = camera.cameraInfo
+                    cameraControl.setLinearZoom(0.5f)
+                } catch (exc: Exception) {
+                    exc.printStackTrace()
+                }
+            },
+            ContextCompat.getMainExecutor(this)
+        )
+    }
+
+    private fun requestAllPermissions() {
+        multiPermissionCallback.launch(REQUIRED_PERMISSIONS)
+    }
+
+    private fun allPermissionsGranted() = REQUIRED_PERMISSIONS.all {
+        ContextCompat.checkSelfPermission(this, it) == PackageManager.PERMISSION_GRANTED
+    }
+
+    private fun aspectRatio(width: Int, height: Int): Int {
+        val previewRatio = max(width, height).toDouble() / min(width, height)
+        if (abs(previewRatio - RATIO_4_3_VALUE) <= abs(previewRatio - RATIO_16_9_VALUE)) {
+            return AspectRatio.RATIO_4_3
+        }
+        return AspectRatio.RATIO_16_9
+    }
+
+    private fun initAnalyzer(screenAspectRatio: Int, rotation: Int): UseCase {
+        return ImageAnalysis.Builder().setTargetAspectRatio(screenAspectRatio)
+            .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+            .setTargetRotation(rotation).build().also {
+                it.setAnalyzer(
+                    executor,
+                    BarcodeAnalyser { barcode ->
+                        if (processingBarcode.compareAndSet(false, false)) {
+                            onBarcodeDetected(barcode)
+                        }
+                    }
+                )
+            }
+    }
+
+    private fun onBarcodeDetected(barcodes: List<Barcode>) {
+        if (barcodes.isNotEmpty() && barcodeViewModel.uiState.value !is UiState.Success) {
+            barcodeViewModel.postBarcode(barcodes[0].rawValue!!)
+        }
+    }
+
+    private fun initCloseBtnClickListener() {
+        binding.btnBarcodeClose.setOnClickListener {
+            finish()
+        }
+    }
+
+    private fun initHardDetectedClickListener() {
+        binding.llBarcodeHardDetected.setOnClickListener {
+            /* 검색뷰 intent 구현 예정 */
+        }
+    }
+
+    private fun collectUiState() {
+        repeatOnStarted {
+            barcodeViewModel.uiState.collect { uiState ->
+                uiState.onSuccess {
+                    Intent(this, CreateUpdateBookActivity::class.java).apply {
+                        putExtra(CREATE, bookData)
+                        putExtra(LOCATION, CREATE)
+                    }.also { intent ->
+                        startActivity(intent)
+                        finish()
+                    }
+                }.onFailed {
+                    /* 에러 다이얼로그 구현 */
+                }
+            }
+        }
```

Comment:
```suggestion
            barcodeViewModel.uiState
                .flowWithLifecycle(lifecycle)
                .collect { uiState ->
                    uiState.onSuccess {
                        Intent(this, CreateUpdateBookActivity::class.java).apply {
                            putExtra(CREATE, bookData)
                            putExtra(LOCATION, CREATE)
                        }.also { intent ->
                            startActivity(intent)
                            finish()
                        }
                    }.onFailed {
                        /* 에러 다이얼로그 구현 */
                    }
                }
```

Reply:
> ```suggestion
> 
>             barcodeViewModel.uiState
> 
>                 .flowWithLifecycle(lifecycle)
> 
>                 .collect { uiState ->
> 
>                     uiState.onSuccess {
> 
>                         Intent(this, CreateUpdateBookActivity::class.java).apply {
> 
>                             putExtra(CREATE, bookData)
> 
>                             putExtra(LOCATION, CREATE)
> 
>                         }.also { intent ->
> 
>                             startActivity(intent)
> 
>                             finish()
> 
>                         }
> 
>                     }.onFailed {
> 
>                         /* 에러 다이얼로그 구현 */
> 
>                     }
> 
>                 }
> 
> ```

이것은 아직 observe가 익숙한 저에게 collect를 observe문법처럼 쉽게 보이게 하고 싶어서 달았습니다!!!